Packs/JoeSecurity/Integrations/JoeSecurityV2/JoeSecurityV2.py

```diff
@@ -0,0 +1,527 @@
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``JoeSandbox``
+    :param JoeSandbox: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    client.server_online()
+    return 'ok'
+
+
+def is_online_command(client: Client) -> CommandResults:
+    """
+         Check if the Joe Sandbox analysis back end is online or in maintenance mode.
+
+         Args:
+            client: (Client) The client class.
+
+         Returns:
+             result: (CommandResults) The CommandResults object .
+    """
+    result = client.server_online()
+    status = 'online' if result.get('online') else 'offline'
+    return CommandResults(readable_output=f"Joe server is {status}")
+
+
+def list_analysis_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+    """
+        List all analyses in descending order, starting with the most recent.
+        The result may be sliced by page and page size or by limit. (default 50)
+
+         Args:
+            client: (Client) The client class.
+            args: (Dict(str, any)) The commands arguments.
+
+         Returns:
+             result: (CommandResults) The CommandResults object.
+    """
+
+    filtered_pages = pagination(args, client.analysis_list_paged())
+    analysis_result_ls = client.analysis_info_list(web_ids=[entry.get('webid') for entry in filtered_pages])
+    return build_analysis_command_result(client, analysis_result_ls)
+
+
+def download_report_command(client: Client, args: Dict[str, Any]) -> Dict[str, Any]:
+    """
+        Download a resource belonging to a report. This can be the full report, dropped binaries, etc.
+
+         Args:
+            client: (Client): The client class.
+            args: (Dict(str, any)): The commands arguments.
+
+         Returns:
+             result: (Dict(str, Any)): The fileResult object.
+    """
+    web_id = args.get('webid')
+    report_type = args.get('type')
+
+    result = client.analysis_download(webid=web_id, type=report_type)[1]
+
+    return fileResult(f'{web_id}_report.{report_type}', result, EntryType.ENTRY_INFO_FILE)
+
+
+def download_sample_command(client: Client, args: Dict[str, Any]) -> Dict[str, Any]:
+    """
+        Download a sample.
+
+         Args:
+            client: (Client): The client class.
+            args: (Dict(str, any)): The commands arguments.
+
+         Returns:
+             result: (Dict(str, Any)): The fileResult object.
+    """
+    web_id = args.get('webid')
+
+    result = client.analysis_download(webid=web_id, type='sample')[1]
+    return fileResult(f'{web_id}.dontrun', result)
+
+
+def search_command(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
```

Comment:
why list?

Reply:
Because `build_analysis_command` result return type is a List and we want to be consistent.
Also, the result can be a list of running on the same file, url, etc..

Reply:
joe-search does not return indicators (it's not a reputation command), so it should return a single command result object with all the list in it.